### PR TITLE
SUP-1293 Generate docs from code

### PR DESCRIPTION
**Pull request by jradtilbrook**
## PR checklist:
- [x] `docs/` updated
  - [x] add import examples
  - [x] update examples with comments to explain them
- [x] tests added
- [x] an example added to `examples/` (useful to demo new field/resource)
- [x] `CHANGELOG.md` updated with pending release information

### CODE REVIEW

**Comment by jradtilbrook** [comment]
Honestly, there isnt much point reviewing this! 😰

File: buildkite/data_source_team.go [hunk 1]
```diff
@@ -46,10 +47,13 @@ func (t *teamDatasource) Metadata(ctx context.Context, req datasource.MetadataRe
 
 func (t *teamDatasource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
 	resp.Schema = schema.Schema{
-		MarkdownDescription: "This datasource allows you to get a team from Buildkite.",
+		MarkdownDescription: heredoc.Doc(`
+			Use this data source to retrieve a team by slug or id. You can find out more about clusters in the Buildkite
+			[documentation](https://buildkite.com/docs/pipelines/permissions).
```
Comment (james2791) [inline buildkite/data_source_team.go:52]: This docs link references clusters when this is around teams? 😉

**Review by james2791 — CHANGES_REQUESTED** [review]

File: buildkite/data_source_team.go (hunk 1 above)
Comment (jradtilbrook) [reply buildkite/data_source_team.go:52]: Lol yep good eyes!

File: buildkite/data_source_team.go [hunk 2]
```diff
@@ -46,10 +47,13 @@ func (t *teamDatasource) Metadata(ctx context.Context, req datasource.MetadataRe
 
 func (t *teamDatasource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
 	resp.Schema = schema.Schema{
-		MarkdownDescription: "This datasource allows you to get a team from Buildkite.",
+		MarkdownDescription: heredoc.Doc(`
+			Use this data source to retrieve a team by slug or id. You can find out more about clusters in the Buildkite
```
Comment (jradtilbrook) [inline buildkite/data_source_team.go:51]:
```suggestion
			Use this data source to retrieve a team by slug or id. You can find out more about teams in the Buildkite
```

File: buildkite/data_source_team.go (hunk 1 above)
Comment (james2791) [reply buildkite/data_source_team.go:52]: Honestly it was a tough find 👍

**Review by james2791 — DISMISSED** [review]
🚀

**Comment by jradtilbrook** [comment]
Ive created this registry off of this branch https://registry.terraform.io/providers/jradtilbrook/greatest/latest (I forked the repo).

So we can view the docs there and review things from the rendered output.

I also tested the semver stuff. The registry shows the latest release docs but installing it is not possible unless you specify the exact version. Eg `2.0.0-beta` will install, but `~> 2.0` will not. And that is what we want for the v1 release

**Review by james2791 — APPROVED** [review]

**Review by james2791 — APPROVED** [review]

**Pull request merged**